packages/opencode/src/util/diagnostic.ts

```
@@ -0,0 +1,212 @@
+): Array<{ diagnostic: Diagnostic; serverID: string }> {
+  if (!Array.isArray(diagnostics) || typeof severity !== "number") {
+    return []
+  }
+
+  return diagnostics.filter((item) => validateDiagnosticItem(item) && item.diagnostic.severity === severity)
+}
+
+/**
+ * Format diagnostics with server grouping and separators
+ */
+export function formatDiagnosticsWithServers(diagnostics: Array<{ diagnostic: Diagnostic; serverID: string }>): string {
```

Biome (2.1.2)

[error] Shouldn't redeclare 'formatDiagnosticsWithServers'. Consider to delete it or rename it.

'formatDiagnosticsWithServers' is defined here:


(lint/suspicious/noRedeclare)